src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,596 @@
+        data_version_filter = [(0, Clusters.BasicInformation, data_version)]
+        read_request_2 = await self.default_controller.ReadAttribute(
+            self.dut_node_id, [(0, Clusters.Objects.BasicInformation.Attributes.NodeLabel)], dataVersionFilters=data_version_filter)
+        data_version_filter_2 = [(0, Clusters.BasicInformation, data_version-1)]
+        read_request_3 = await self.default_controller.ReadAttribute(
+            self.dut_node_id, [(0, Clusters.Objects.BasicInformation.Attributes.NodeLabel)], dataVersionFilters=data_version_filter_2)
+        asserts.assert_equal(int(read_request_3[0][Clusters.Objects.BasicInformation][Clusters.Objects.BasicInformation.Attributes.NodeLabel]),
+                             data_version_value, f"Data version does not equal {data_version_value}")
+
+        # Step 27
+
+        # TH sends a Read Request Message to the DUT to read any supported attribute/wildcard on a particular cluster say A with the DataVersionFilter Field not set.
+        # DUT sends back the attribute value with the DataVersion of the cluster A.
+        # TH sends a Read Request Message to read any supported attribute/wildcard on cluster A and any supported attribute/wildcard on another cluster B.
+        # DataVersionList field should only contain the DataVersion of cluster A.
+
+        # Verify that the DUT sends a report data action with the attribute value from the cluster B to the TH.
+        # Verify that the DUT does not send the attribute value from cluster A.
+
+        self.print_step(
+            27, "Send the Read Request Message to read any supported attribute/wildcard on a particular cluster say A with the DataVersionFilter Field not set")
+        read_request_27_1_1 = await self.default_controller.ReadAttribute(self.dut_node_id, [Clusters.Objects.Descriptor.Attributes.ServerList])
+
+        data_version_1 = read_request_27_1_1[0][Clusters.Descriptor][Clusters.Attribute.DataVersion]
+        read_request_27_2_1 = await self.default_controller.ReadAttribute(self.dut_node_id, [(0, Clusters.BasicInformation.Attributes.NodeLabel)])
+        data_version_2 = read_request_27_2_1[0][Clusters.BasicInformation][Clusters.Attribute.DataVersion]
+
+        asserts.assert_not_equal(data_version_1, data_version_2)
+
+        # Step 28
+
+        # TH sends a Read Request Message to the DUT to read something(Attribute) which is larger than 1 MTU(1280 bytes) and per spec can be chunked +
+
+        # Verify on the TH that the DUT sends a chunked data message with the SuppressResponse field set to False for all the messages except the last one.
+        # Verify the last chunked message sent has the SuppressResponse field set to True.
+
+        self.print_step(
+            28, "Send the Read Request Message to read something(Attribute) which is larger than 1 MTU(1280 bytes) and per spec can be chunked +")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, "*")
```

Comment:
The checks here aren't implemented. Though give me a second on this one - I would like to check with the camera team to see if this is actually correct. It has a PICS guard and I don't think that's correct either.

Reply:
What should I do with the PICS guard?